### ISSUE TITLE
Merge with gulp-handlebars?

### ISSUE THREAD

**Issue by lazd** [mattma/gulp-htmlbars#2]
It seems this plugin and `gulp-handlebars` share the same goal and even include some of the same documentation.

I think they should be combined. I'll gladly accept a pull request to add the missing functionality to `gulp-handlebars`.

Thoughts?

**Comment by mattma** [comment]
Yes. @lazd 

I totally agree with you on this. I am more than happy to merge two into one (gulp-handlebars onwards)

The whole reason that I created this project is because [ember-rocks](https://github.com/mattma/ember-rocks) stayed with my forked branch for a while, since it fetched the Tarball instead of getting from NPM registry. It does not seem to be a solid solution for users. On top of that, HTMLBars released and replaced the handlebars for some cents. I make the call to start a project for stability. And now, you have time back on this project. It is awesome. We will work together to make it into one. [This is how I use it in Ember-rocks](https://github.com/mattma/ember-rocks/blob/master/src/skeletons/core/gulpfile.js#L158).

What is your plan on merging?

**Comment by lazd** [comment]
Hey @mattma,you're definitely the Ember expert between the two of us, so if you have this working and solved the issues mentioned in https://github.com/swelham/gulp-ember-templates/issues/3, send along a PR and let's get it in!

As long as basic Handlebars template compilation still works, I'm fine with additional changes to support Ember inside of `gulp-handlebars`. Let me know if there is anything I can do to help you, I'll definitely test Handlebars 1/2/3 manually and will write some automated tests so we can be sure that everything is behaving as expected across versions.

I doubt this will help since you seem to have already thoroughly explored the issues, but I did find [some interesting problems](https://github.com/swelham/gulp-ember-templates/issues/3#issuecomment-78539996) and halfway solved them in [this branch](https://github.com/lazd/gulp-handlebars/tree/ember-template-compiler).

Let's keep in mind what @rwjblue said through this process:

> As of Ember 1.10 and above, you must compile templates with the template compiler published with your specific build of Ember.
> See http://emberjs.com/blog/2015/02/05/compiling-templates-in-1-10-0.html for more details.

Thanks!

**Comment by mattma** [comment]
@lazd sure. For Ember 1.10 and above, I am currently passing [an option](https://github.com/mattma/ember-rocks/blob/master/src/skeletons/core/gulpfile.js#L160) to the `gulp-htmlbars` so that each Ember version would use a different compiler.

I saw that you released a newer version v4.0.0, and my previous [PR](https://github.com/lazd/gulp-handlebars/pull/44) is still pending. Should I start a new PR process with the changes?

**Comment by lazd** [comment]
Let's get your previous PR in, then we can build off of it for the HTML bars stuff. I'll go ahead and merge right now.

**Comment by lazd** [comment]
I went ahead and merged. I've also added you as a collaborator on `gulp-handlebars`. If you're ready, go ahead and put together a branch that adds in all the functionality we need for Ember and HTMLBars, send a pr against master, and we'll fast track it for merge and release. Thanks!

**Comment by mattma** [comment]
It is my honor to work along side with you on `gulp-handlebars`. I will do what you just said. Thank you! @lazd

**Comment by lazd** [comment]
Likewise! Thank you, @mattma!

**Comment by mattma** [comment]
Once the migration is done, I will mark the deprecation notice in this repo and redirect people to `gulp-handlebars`

**Comment by erg** [comment]
How is this going? I'm getting confused with which package to use. The README still says end of March, which has come and gone.

Cheers!

**Comment by lazd** [comment]
@erg, see https://github.com/lazd/gulp-handlebars/pull/51#issuecomment-111773374

**Comment by mattma** [comment]
@erg Thanks. @lazd 

Using `gulp` with `ember` is not going to be the main stream. Ember has officially announced which tech stack that Ember team is going to support. See [here](http://emberjs.com/blog/2015/06/16/ember-project-at-2-0.html) 

The wild package dependency combination of Ember project are countless. With a new release of Ember 2.0, along with new default "glimmer" which is the 3rd version of Ember rendering engine, and it is almost completely rewrites, departed from original handlebars implementation due to its natural of Virtual DOM diff logic. It is fairly hard to create a `gulp` plugin to handle 3 different version of Ember rendering engine. 

I suggest that `gulp-handlebars` to keep its own functionality and drop the support for `Ember`. Under the current rapid Ember development, angle bracket component is going to land in July, which is another major changes from the previous V2 of HTMLBars. 

I would be happy to answer any Ember related question on Gulp and Handlebars.
